sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobCheckpointData.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.DataMovement.Blobs
+{
+    /// <summary>
+    /// Base class for Blob source and destination checkpoint data
+    /// which contains shared fields.
+    /// </summary>
+    internal abstract class BlobCheckpointData : StorageResourceCheckpointData
```

Comment:
That's sort of a bigger ask since this pattern exists for all fields in all of our schema files, I was just matching what we have. If we feel strongly we want to make this change, we can, but probably something for a separate PR so we can do it wholistically.